examples/fsharp.extraction.targets

```diff
@@ -63,8 +63,8 @@
         <!-- If FSTAR_HOME is not defined then assume that fstar.exe is available on the path -->
         <FSTAR_EXE Condition="'$(FSTAR_HOME)' == ''">fstar.exe</FSTAR_EXE>
         <FSTAR_EXE Condition="'$(FSTAR_HOME)' != ''">$(FSTAR_HOME)\bin\fstar.exe</FSTAR_EXE>
-        <CALL_PROC_EXE Condition="'$(FSTAR_HOME)' == ''">CallProc.exe</CALL_PROC_EXE>                
-        <CALL_PROC_EXE Condition="'$(FSTAR_HOME)' != ''">$(FSTAR_HOME)\bin\CallProc.exe</CALL_PROC_EXE>        
+        <CALL_PROC_EXE Condition="'$(FSTAR_HOME)' == '' and '$(OS)' != 'Windows_NT' ">CallProc.exe</CALL_PROC_EXE>                
+        <CALL_PROC_EXE Condition="'$(FSTAR_HOME)' != '' and '$(OS)' != 'Windows_NT'">$(FSTAR_HOME)\bin\CallProc.exe</CALL_PROC_EXE>        
```

Comment:
Why do we need this additional check, my understanding (from #2512) is that `CallProc.exe` works on both Windows and Linux.

Reply:
I do have issues with `CallProc.exe` in one of my setups. Not sure why I have that, but I belive that there no need to have separate apps until nescessary. If that's again only for old Mono case, then probably it can be removed completely. 

Because of long delay, I'm relatively forgot why specifically I remove that. What I wrote is relative hints which I have. I have to replicate work to provide better explanation.

Reply:
Okay. I seems to be understand what issue I was hitting. right now I do not see how `CallProc.exe` is built in the repo, and I do see the case for complete removal of dependency on Msbuild or XBuild, since it seems the only available modern (>2017) toolchain is .NET SDK and that mean we can use simply `dotnet build XXX.csproj` and `dotnet build YYY.csproj` without too much thinking.

I'm still running errands on the internet to better understand that Mono ecosystem, but what I see is that Mono is runtime environment only, no dev support provided and .NET is the future.  

There options to use MsBuild as separate executable, but on Ubuntu 20.04 this is on preview feed, and you know, default on Linux is not a Mono right now.

Please correct me if I'm wrong somewhere in my assumptions.

Reply:
Ignore, I otherthinking here about `CallProc.exe`. All things about Mono still holds. You can read https://github.com/FStarLang/FStar/pull/2513#issuecomment-1085516293 about additional rationale for xbuild/msbuild drop

Reply:
I didn't really catch up with all of the discussions yet, but do we still need the CallProc.exe at all? If so, do we understand why?

Reply:
I do not know right now. I just don't want to rock the boat, and remove it if needed in separate PR.

Personally I think think that file is not needed, but I do not know if team use that somehow for having uniform line endings in log (?) or whatever.

Reply:
Afair, this was added by @nikswamy due to issues on CI as without it fstar.exe was called as `mono fstar.exe`, when called through msbuild targets.

Now that mono is getting removed it shouldn't be a problem anymore so it could be removed completely.

@nikswamy WDYT?

Reply:
If it is possible to remove that CallProc, I'll be all for it!